R/init.R

```diff
@@ -222,27 +186,242 @@ init <- function(data,
+    lifecycle::deprecate_warn(
+      when = "0.15.3",
+      what = "init(header)",
+      details = paste(
+        "Use `modify_header()` on the teal app object instead."
+      )
+    )
+  } else {
+    header <- tags$p()
+  }
+  if (!is.null(footer)) {
+    checkmate::assert_multi_class(footer, c("shiny.tag", "shiny.tag.list", "html", "character"))
+    lifecycle::deprecate_warn(
+      when = "0.15.3",
+      what = "init(footer)",
+      details = paste(
+        "Use `modify_footer()` on the teal app object instead."
+      )
+    )
+  } else {
+    footer <- tags$p()
+  }
+
+  # argument transformations
+  ## `modules` - landing module
+  landing <- extract_module(modules, "teal_module_landing")
+  if (length(landing) == 1L) {
+    landing_popup_server <- landing[[1L]]$server
+    modules <- drop_module(modules, "teal_module_landing")
+    lifecycle::deprecate_warn(
+      when = "0.15.3",
+      what = "landing_popup_module()",
+      details = paste(
+        "landing_popup_module() is deprecated.",
+        "Use add_landing_popup() on the teal app object instead."
+      )
+    )
+  } else if (length(landing) > 1L) {
+    stop("Only one `landing_popup_module` can be used.")
+  }
+
   # Note: UI must be a function to support bookmarking.
   res <- list(
-    ui = function(request) {
+    ui = function(request, ...) {
       ui_teal(
-        id = ns("teal"),
+        id = "teal",
         modules = modules,
         title = title,
         header = header,
         footer = footer
       )
     },
     server = function(input, output, session) {
-      if (!is.null(landing_popup)) {
-        do.call(landing_popup$server, c(list(id = "landing_module_shiny_id"), landing_popup$server_args))
+      if (!is.null(landing_popup_server)) {
+        do.call(landing_popup_server, c(list(id = "landing_module_shiny_id")))
       }
-      srv_teal(id = ns("teal"), data = data, modules = modules, filter = deep_copy_filter(filter))
+      srv_teal(id = "teal", data = data, modules = modules, filter = deep_copy_filter(filter))
     }
   )
 
   logger::log_debug("init teal app has been initialized.")
 
   res
 }
+
+#' Add a custom title to the app
+#'
+#' @param title The title to add
+#' @return The modified app object
+#' @export
+#' @examples
+#' app <- init(
+#'   data = teal_data(IRIS = iris, MTCARS = mtcars),
+#'   modules = modules(example_module())
+#' ) |>
+#'   modify_title("Custom title")
+#'
+#' shinyApp(app$ui, app$server)
+modify_title <- function(
+    app,
+    title = "teal app",
+    favicon = "https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/PNG/nest.png") {
+  res <- app
+  res$ui <- function(request, ...) {
+    args <- list(...)
+    args$title <- tags$div(
+      id = "teal-title",
+      build_app_title(title, favicon)
+    )
+    ui_tq <- htmltools::tagQuery(do.call(app$ui, c(list(request = request), args)))
+
+    ui_tq$find("#teal-title")$replaceWith(args$title)$allTags()
+  }
```

Comment:
`...` seems to be redundant everywhere as in `init` it is not consumed.

```suggestion
  res$ui <- function(request) {
    title <- tags$div(
      id = "teal-title",
      build_app_title(title, favicon)
    )
    ui_tq <- htmltools::tagQuery(app$ui(request = request))

    ui_tq$find("#teal-title")$replaceWith(title)$allTags()
  }
```